### PR TITLE
Frontend export listing and editing

### PR DESCRIPTION
**Pull request by benrudolph**
@NoahCarnahan @snopoke this one is a bit of a beast, can be :dolphin: or :office:. i think i'll mark up the PR to make it more clear. the main source of complexity came from the fact that we need to maintain the old state of the export when updating to a new schema. i.e. you select some questions, those get deleted, you open the export and those are still selected but now marked as deleted.

### CODE REVIEW

File: corehq/couchapps/export_instances_by_domain/views/view/map.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+function(doc) {
```
Comment (dimagimon) [inline corehq/couchapps/export_instances_by_domain/views/view/map.js]: Missing name in function declaration.

File: corehq/apps/export/models/new.py [hunk 2]
```diff
@@ -105,32 +105,44 @@ def create_default_from_export_item(group_schema_path, item, build_ids_and_versi
...
         is_deleted = True
         for app_id, version in build_ids_and_versions.iteritems():
-            if item.last_occurrences.get(app_id) == version:
+            if self.item.last_occurrences.get(app_id) == version:
                 is_deleted = False
                 break
+        return is_deleted
 
-        is_main_table = group_schema_path == MAIN_TABLE
+    def update_properties_from_build_ids(self, build_ids_and_versions):
```
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: this basically ensures that the column is "up-to-date" with any changes that might've happened

File: corehq/apps/export/models/new.py [hunk 3]
```diff
@@ -229,43 +247,101 @@ class Meta:
...
+            table = instance.get_table(group_schema.path) or TableConfiguration(
                 path=group_schema.path,
                 name=instance.defaults.get_default_table_name(group_schema.path),
                 display_name=instance.defaults.get_default_table_name(group_schema.path),
                 selected=instance.defaults.default_is_table_selected(group_schema.path),
             )
-            table.columns = map(
-                lambda item: ExportColumn.create_default_from_export_item(
+            columns = []
+            for item in group_schema.items:
```
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: this loop is probably the most complex. essentially, if the column exists in the old export, grab it. update the item and other props. there's one edge case that i haven't accounted for yet. we distinguish `ExportItem`s by path _and_ doctype meaning we can have a question with the same path if they've changed it from a scalar to a multiple choice question. in this case get_column will just return the first one it finds... probably not the best. we could distinguish by path and doc_type, but then we have the potential of having two column with the same exact name. i believe the reason we keep around multi choice question is to ensure we still have access to the options. though now thinking about it, i'm not sure it's necessary

File: corehq/apps/export/views.py [hunk 4]
```diff
@@ -1127,6 +1209,8 @@ def bulk_download_url(self):
     @memoized
     def get_saved_exports(self):
         exports = FormExportSchema.get_stale_exports(self.domain)
+        new_exports = get_form_export_instances(self.domain)
```
Comment (benrudolph) [inline corehq/apps/export/views.py]: surprisingly this kind of just works with only a few minor tweakings. i haven't tried daily saved exports, but i don't imagine it'd be too hard to get those working

File: corehq/apps/export/models/new.py [hunk 5]
```diff
@@ -105,32 +105,44 @@ def create_default_from_export_item(group_schema_path, item, build_ids_and_versi
 
         :param group_schema_path: The path of the group_schema that the item belongs to
         :param item: An ExportItem instance
-        :param app_version: The app version that the column export is for
+        :param build_ids_and_versions: A dictionary of build ids that map to versions of that represents the
```
Comment (snopoke) [inline corehq/apps/export/models/new.py]: `versions of that represents the` what :grey_question:

File: corehq/apps/export/models/new.py [hunk 6]
```diff
@@ -105,32 +105,44 @@ def create_default_from_export_item(group_schema_path, item, build_ids_and_versi
...
                 is_deleted = False
                 break
+        return is_deleted
 
-        is_main_table = group_schema_path == MAIN_TABLE
+    def update_properties_from_build_ids(self, build_ids_and_versions):
+        """
+        This regenerates properties based on new build ids/versions
+        :param build_ids_and_versions: A dictionary of build ids that map to versions of that represents the
+        most recent state of the app(s) in the domain
```
Comment (snopoke) [inline corehq/apps/export/models/new.py]: `versions of that represents` grammar :book:

Comment (benrudolph) [inline corehq/apps/export/models/new.py]: lol that's pretty painful to parse in the head. will update!

File: corehq/apps/export/models/new.py (hunk 3 above)
Comment (snopoke) [inline corehq/apps/export/models/new.py]: I don't think it's that big a deal:
* single -> multi: any 'extra' options will get put in the 'other' column
* multi-> single: all values in a single column

I think the only issue is if the user started with a single select and then the app changed to a multi-select then their custom export will stay as a single select. If they don't want they then they'd have to remove the column, save the export and then add the column back (assuming that would work).

File: corehq/apps/export/models/new.py [hunk 7]
```diff
@@ -315,7 +394,7 @@ def get_default_table_name(table_path):
 
     @staticmethod
     def get_default_instance_name(schema):
-        return u'{}: {}'.format(schema.case_type, datetime.now().strftime('%Y-%M-%d'))
+        return u'{}: {}'.format(schema.case_type, datetime.now().strftime('%Y-%m-%d'))
```
Comment (snopoke) [inline corehq/apps/export/models/new.py]: `2016-59-02` :eyes:

File: corehq/apps/export/views.py [hunk 8]
```diff
@@ -153,7 +162,7 @@ def report_class(self):
             }
             return base_views[self.export_type]
         except KeyError:
-            raise SuspiciousOperation
+            raise SuspiciousOperation('Attempted to access list view {}'.format(self.export_type))
```
Comment (snopoke) [inline corehq/apps/export/views.py]: :ghost:

File: corehq/apps/export/models/new.py (hunk 7 above)
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: yeah haha took me a while to realize i wasn't looking at the date

File: corehq/apps/export/views.py (hunk 4 above)
Comment (snopoke) [inline corehq/apps/export/views.py]: nice

File: corehq/apps/export/models/new.py (hunk 6 above)
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: https://github.com/dimagi/commcare-hq/commit/a744af5cc73a9e2cc9472ffdabd5cfa5163e822a
 this was actually very confusing. i'm passing in the app ids that map to the latest build versions, yet all the variables say build ids...

File: corehq/apps/export/views.py [hunk 9]
```diff
@@ -1251,7 +1345,7 @@ def fmt_export_data(self, export):
             'addedToBulk': False,
             'exportType': export.type,
             'emailedExports': emailed_exports,
-            'editUrl': reverse(EditCustomCaseExportView.urlname,
+            'editUrl': reverse(edit_view.urlname,
```
Comment (snopoke) [inline corehq/apps/export/views.py]: What happens if have the toggle set but you click on an old export? Will this break?

File: corehq/apps/export/models/new.py (hunk 6 above)
Comment (snopoke) [inline corehq/apps/export/models/new.py]: :confused:

File: corehq/apps/export/views.py (hunk 9 above)
Comment (benrudolph) [inline corehq/apps/export/views.py]: absolutely. but this is the next piece of functionality that i'm working on. essentially it takes the id and if it finds the old schema, it takes it and converts it to the new. probably easier said than done

**Comment by snopoke** [comment]
looking good

**Comment by snopoke** [comment]
LGTM though I don't see a build

**Comment by benrudolph** [comment]
it's into another branch, OK to merge

**Comment by snopoke** [comment]
:evergreen_tree:

**Pull request merged**